File: app/src/main/java/com/tt/handsomeman/model/Job.java
package com.tt.handsomeman.model;

import com.tt.handsomeman.HandymanApp;
import com.tt.handsomeman.R;
import com.tt.handsomeman.util.TimeCount;

import java.io.Serializable;
import java.util.Calendar;
import java.util.Date;

public class Job implements Serializable {

    private Integer id;
    private Integer categoryId;
    private String title;
    private Date createTime;
    private String detail;
    private Integer budgetMin;
    private Integer budgetMax;
    private Date deadline;
    private String location;
    private Double lat;
    private Double lng;
    private Integer customerId;

    public Integer setCreateTimeBinding() {
        Calendar now = Calendar.getInstance();
        return TimeCount.getHoursBetween(this.createTime, now.getTime());
    }

    public String setBudgetRange() {
        return HandymanApp.getInstance().getString(R.string.budget_range, this.budgetMin, this.budgetMax);
    }

    public Integer setDeadlineBinding() {
        Calendar now = Calendar.getInstance();
        return Math.max(TimeCount.getDaysBetween(now.getTime(), this.deadline), 0);
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Integer getCategoryId() {
        return categoryId;
    }

    public void setCategoryId(Integer categoryId) {
        this.categoryId = categoryId;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public Date getCreateTime() {
        return createTime;
    }

    public void setCreateTime(Date createTime) {
        this.createTime = createTime;
    }

    public Date getDeadline() {
        return deadline;
    }

    public void setDeadline(Date deadline) {
        this.deadline = deadline;
    }

    public String getDetail() {
        return detail;
    }

    public void setDetail(String detail) {
        this.detail = detail;
    }

    public Integer getBudgetMin() {
        return budgetMin;
    }

    public void setBudgetMin(Integer budgetMin) {
        this.budgetMin = budgetMin;
    }

    public Integer getBudgetMax() {
        return budgetMax;
    }

    public void setBudgetMax(Integer budgetMax) {
        this.budgetMax = budgetMax;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public Double getLat() {
        return lat;
    }

    public void setLat(Double lat) {
        this.lat = lat;
    }

    public Double getLng() {
        return lng;
    }

    public void setLng(Double lng) {
        this.lng = lng;
    }

    public Integer getCustomerId() {
        return customerId;
    }

    public void setCustomerId(Integer customerId) {
        this.customerId = customerId;
    }
}

File: app/src/main/java/com/tt/handsomeman/model/Job.kt
package com.tt.handsomeman.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

class Job(@field:SerializedName("createTime")
          @field:Expose
          var createTime: Date?, @field:SerializedName("title")
          @field:Expose
          var title: String?, @field:SerializedName("budgetMin")
          @field:Expose
          var budgetMin: Int?, @field:SerializedName("budgetMax")
          @field:Expose
          var budgetMax: Int?, @field:SerializedName("deadline")
          @field:Expose
          var deadline: Date?, @field:SerializedName("location")
          @field:Expose
          var location: String?) {

    @SerializedName("id")
    @Expose
    var id: Int? = null
    @SerializedName("categoryId")
    @Expose
    var categoryId: Int? = null
    @SerializedName("detail")
    @Expose
    var detail: String? = null
    @SerializedName("lat")
    @Expose
    var lat: Double? = null
    @SerializedName("lng")
    @Expose
    var lng: Double? = null
    @SerializedName("bidMin")
    @Expose
    var bidMin: Int? = null
    @SerializedName("bidMax")
    @Expose
    var bidMax: Int? = null
    @SerializedName("interviewing")
    @Expose
    var interviewing: Int? = null
    @SerializedName("handymanId")
    @Expose
    var handymanId: Int? = null
    @SerializedName("accountId")
    @Expose
    var accountId: Int? = null
    @SerializedName("status")
    @Expose
    var status: String? = null

    fun setCreateTimeBinding(createTime: Date): String {
        val myFormat = "h" //In which you need put here
        val sdf = SimpleDateFormat(myFormat, Locale.US)
        return sdf.format(createTime) + " hour(s) ago"
    }

    fun setBudgetRange(budgetMin: Int?, budgetMax: Int?): String {
        return "$ $budgetMin - $$budgetMax"
    }

    fun setLocationBinding(location: String): String {
        var location = location
        if (location.length > 10) {
            location = location.substring(0, 11) + "..."
        }
        return location
    }

    fun setDeadlineBinding(deadline: Date): String {
        val myFormat = "dd" //In which you need put here
        val sdf = SimpleDateFormat(myFormat, Locale.US)
        return sdf.format(deadline) + " day(s) left"
    }
}
